openid/jwksgettermock_test.go

```diff
@@ -26,13 +27,16 @@ type getJwksResponse struct {
 }
 
 func (c *jwksGetterMock) getJwkSet(r *http.Request, url string) (jose.JsonWebKeySet, error) {
-	c.Calls <- &getJwksCall{url}
+	c.Calls <- &getJwksCall{r, url}
 	gr := (<-c.Calls).(*getJwksResponse)
 	return gr.jwks, gr.err
 }
 
-func (c *jwksGetterMock) assertGetJwks(url string, jwks jose.JsonWebKeySet, err error) {
+func (c *jwksGetterMock) assertGetJwks(req *http.Request, url string, jwks jose.JsonWebKeySet, err error) {
 	call := (<-c.Calls).(*getJwksCall)
+	if req == nil || call.req != req {
```

Comment:
req != nil so you don't need to pass a request in all test but just a single one that actually intends to validate this.